opm/parser/eclipse/EclipseState/SimulationConfig/tests/ThresholdPressureTest.cpp

```diff
@@ -181,13 +181,10 @@ static DeckPtr createDeck(const ParseContext& parseContext , const std::string&
     return parser.parseString(input , parseContext);
 }
 
-static Eclipse3DProperties getEclipseProperties(const Deck& deck)
-{
-    static const EclipseGrid eclipseGrid( 3, 3, 3 );
-    return Eclipse3DProperties( deck, TableManager( deck ), eclipseGrid );
+static Eclipse3DProperties getEclipseProperties(const Deck& deck) {
+    return Eclipse3DProperties(deck, *new TableManager(deck), *new EclipseGrid(3, 3, 3));
```

Comment:
@atgeirr I didn't rebase, but I did amend the change into the previous commit.

When it comes to the memory leak, I've added a comment specifying that it leaks.  It's a tradeoff between leaking one place, and needing something like 10--16 new references in one of the tests, and a couple of new references in the other tests.  If we prefer the latter, I can do it, but it really clutters up the tests, and make them much less readable, thus also less useful as guidelines and examples.